### ISSUE TITLE
Contacts duplicated & collected

### ISSUE THREAD

**Issue by Ampli-fier** [deltachat/deltachat-core#32]
**Version Info**
Delta Chat v0.9.1
CyanogenMod 12.1 (5.1.1)

**Description**
I received a contact request that was sent to multiple recipients.

1. Email addresses that were not in my phone book before, are now stored in Delta Chat _contacts_ table (IDs 275 & 276 below).
I can see them in the UIs "New Chat" list.
Is this wanted behaviour, to collect email addresses (of people who did not actively try to contact me)?

2. One of the contacts was in my phone book before.
This person ("Recipient4") had two mail addresses in the phone book entry (IDs 13 & 70 below).
It looks like Delta Chat did not manage to find the match and added another entry (ID 277).

**Logs**
To: Recipient1 <recip1@domain.com>,
 Recipient2 <recip2@domain.com>,
 Recipient3 <recip3@doamin.com>,
 Myself <Ampli-fier@domain.com>, recip4@eu.domain.com,
 Recipient5 <recip5@domain.com>, recip6@domain.com

INSERT INTO "contacts" VALUES(13,'Recipient4','recip4@eu.domain.com',524288,0,0,'','');
INSERT INTO "contacts" VALUES(70,'Recipient4','name@domain.de',524288,0,0,'','');

INSERT INTO "contacts" VALUES(275,'Recipient2','recip2@domain.com',1024,0,0,'','');
INSERT INTO "contacts" VALUES(276,'Recipient3','recip3@domain.com',1024,0,0,'','');
INSERT INTO "contacts" VALUES(277,'','recip4@eu.domain.com',1024,0,0,'','');

**PS**: Thank you for the hard work to turn this great idea into a working app!!

**Comment by r10s** [comment]
to 1: yes, Delta Chat collects addresses contacted by you or your contacts, see https://delta.chat/en/help#which-messages-do-appear-in-delta-chat - however, regarding mailing lists Delta Chat opened too many chats, see https://github.com/deltachat/deltachat-android/issues/120, which we've just fixed.

to 2: this should not happen, two identical e-mail-addresses in the phone book; maybe this is a bug

**Comment by r10s** [comment]
still to 2: At a first look at the source code I cannot imaging, how this can happen, see https://github.com/deltachat/deltachat-core/blob/master/src/mrcontact.c#L239 where the INSERT INTO is executed - just after checking that the email address is not yet in the database.
Are you sure, the two email addresses are 100% the same? Can you reproduce the behavior in a way?

**Comment by Ampli-fier** [comment]
to 2:
In the mean time, the entry (ID 277) is gone, there is a gap in the contacts database (... ,275, 276, 278, 279, ...).
However, I think the entry is now present with the ID 280. And when I compare this mail address with the one I have in my phone book, I see the difference. A minus was removed from the name and I overlooked that.
Sorry for confusion and thanks for checking.

to 1:
Good, that you found a fix for mailing lists. Looking forward to the new version.
